### ISSUE TITLE
Please remove your own `go-*` directories only

### ISSUE THREAD

**Issue by kseistrup** [axllent/wireguard-vanity-keygen#3]
The `Makefile` has:

```sh
        rm -rf /tmp/go-*
```

https://github.com/axllent/wireguard-vanity-keygen/blob/develop/Makefile#L12

However, `wireguard-vanity-keygen` may not be the only go building process running now, and so by removing all `go-*` directories, “our” process may induce collateral damage by removing directories it hasn't created. “We” should only ever remove our own.

**Comment by axllent** [comment]
You are absolutely right - that Makefile was more for my own ease-of-use than intended for others to use. I will try set some time aside soon to migrate to a a proper deployment/build process. If not installing from binary but rather from source, then the intended way of install is `go install github.com/axllent/wireguard-vanity-keygen@latest`

**Comment by kseistrup** [comment]
Thanks.

I came here because I had installed `wireguard-vanity-keygen` from AUR, and noticed that it was using the `Makefile` to install the package:

```sh
build() {
    cd "${srcdir}/${pkgname}-${pkgver}"
    make clean build
}
```

* https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=wireguard-vanity-keygen

What AUR packagers do is of course not your responsibility, but if the `Makefile` only removed its own `go-*` directories it would make it more robust and less prone to surprises for everyone.

I am not very well versed in the go programming language, but would it be possible to specify a “private” temporary directory and use that instead?

E.g., if we hypothetically say that `go` will place its temporary files and directories in `$GO_TMPDIR`, then it could be as simple as (as a shell script, I'm unsure how to incorporate this in a `Makefile`):

```sh
$GO_TMPDIR=$(mktemp -d)
make clean build
rm -rf "$GO_TMPDIR"
```

Cheers.
